ratpack-core/src/test/groovy/ratpack/core/http/client/HttpClientRedirectionSpec.groovy

```diff
@@ -153,6 +153,41 @@ class HttpClientRedirectionSpec extends BaseHttpClientSpec {
     pooled << [true, false]
   }
 
+  def "can follow a relative redirect get request with non-encoded query parameters"() {
```

Comment:
The non-encoded query parameter test cases seem questionable to me.  It's unclear to me what the behavior should be as per the spec.

Reply:
Hmm... Yes, I agree.  I just put that test in to capture the current behavior, which is the same before and after the change here.

Like you indicate, the spec really only speaks of the location header being a URL, and doesn't mention much about the encoded/decoded nature of this value:

https://developer.mozilla.org/en-US/docs/Web/HTTP/Redirections

The current ratpack implementation for query parameters does decode them. This change should only be impacting path parameters.

So... As a test that indicates that this behavior is the "contracted, expected result", it's a bit ambiguous, and probably not so great. But, as a regression test indicating that nothing has changed if future work is done in this are, it has value. 

I kept the test, but put in a comment.  Would that work for you?   I can alter wording if it's not clear as well.

Reply:
Hmm... Yes, I agree.  I just put that test in to capture the current behavior, which is the same before and after the change here.

Like you indicate, the spec really only speaks of the location header being a URL, and doesn't mention much about the encoded/decoded nature of this value:

https://developer.mozilla.org/en-US/docs/Web/HTTP/Redirections

The current ratpack implementation for query parameters does decode them. This change should only be impacting path parameters.

So... As a test that indicates that this behavior is the "contracted, expected result", it's a bit ambiguous, and probably not so great. But, as a regression test indicating that nothing has changed if future work is done in this are, it has value. 

I kept the test, but put in a comment.  Would that work for you?   I can alter wording if it's not clear as well.

Reply:
Yes, that approach is fine with me.